non-linear-solver-spec.json

```diff
@@ -15,7 +15,9 @@
             "LBFGS",
             "LBFGSB",
             "Newton",
-            "box_constraints"
+            "box_constraints",
+            "f_delta",
```

Comment:
Did we decide to put them somewhere, not in the root?

Reply:
ok, so one more level, put them in `should_not_use`?